homeassistant/components/assist_satellite/__init__.py

```diff
@@ -0,0 +1,195 @@
+    "AssistSatelliteState",
+    "AssistSatelliteEntity",
+    "SatelliteConfig",
+    "SatelliteCapabilities",
+    "PipelineRunConfig",
+    "PipelineRunResult",
+]
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORM_SCHEMA = cv.PLATFORM_SCHEMA
+PLATFORM_SCHEMA_BASE = cv.PLATFORM_SCHEMA_BASE
+
+ATTR_WAKE_WORDS = "wake_words"
+ATTR_PROCESS = "process"
+ATTR_ANNOUNCE_TEXT = "announce_text"
+ATTR_MUTED = "is_muted"
+
+SERVICE_WAIT_WAKE = "wait_wake"
+SERVICE_GET_TEXT = "get_text"
+SERVICE_SAY_TEXT = "say_text"
+SERVICE_MUTE_MICROPHONE = "mute_microphone"
+
+
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+    component = hass.data[DOMAIN] = EntityComponent[AssistSatelliteEntity](
+        _LOGGER, DOMAIN, hass
+    )
+    await component.async_setup(config)
+
+    component.async_register_entity_service(
+        name=SERVICE_WAIT_WAKE,
+        schema=cv.make_entity_service_schema(
+            {
+                vol.Required(ATTR_WAKE_WORDS): [cv.string],
+                vol.Optional(ATTR_ANNOUNCE_TEXT): cv.string,
+            }
+        ),
+        func=async_service_wait_wake,
+        required_features=[AssistSatelliteEntityFeature.TRIGGER],
+        supports_response=SupportsResponse.OPTIONAL,
+    )
+
+    component.async_register_entity_service(
+        name=SERVICE_GET_TEXT,
+        schema=cv.make_entity_service_schema(
+            {
+                vol.Optional(ATTR_PROCESS): cv.boolean,
+                vol.Optional(ATTR_ANNOUNCE_TEXT): cv.string,
+            }
+        ),
+        func=async_service_get_text,
+        required_features=[AssistSatelliteEntityFeature.TRIGGER],
+        supports_response=SupportsResponse.OPTIONAL,
+    )
+
+    component.async_register_entity_service(
+        name=SERVICE_SAY_TEXT,
+        schema=cv.make_entity_service_schema(
+            {vol.Required(ATTR_ANNOUNCE_TEXT): cv.string}
+        ),
+        func=async_service_say_text,
+        required_features=[AssistSatelliteEntityFeature.TRIGGER],
+        supports_response=SupportsResponse.NONE,
+    )
+
+    component.async_register_entity_service(
+        name=SERVICE_MUTE_MICROPHONE,
+        schema=cv.make_entity_service_schema({vol.Required(ATTR_MUTED): cv.boolean}),
+        func=async_service_mute_microphone,
+        required_features=[AssistSatelliteEntityFeature.AUDIO_INPUT],
+        supports_response=SupportsResponse.NONE,
+    )
+
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up a config entry."""
+    component: EntityComponent[AssistSatelliteEntity] = hass.data[DOMAIN]
+    return await component.async_setup_entry(entry)
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Unload a config entry."""
+    component: EntityComponent[AssistSatelliteEntity] = hass.data[DOMAIN]
+    return await component.async_unload_entry(entry)
+
+
+async def async_service_wait_wake(
```

Comment:
When these services are added back in a future PR, 
these shouldn't be top-level functions but instead be functions on the Assist Satellite base class and then pass the string of that function to the `register_enttiy_service` function